### PR TITLE
Added validators tests.

### PR DESCRIPTION
**Pull request by KR4K0**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # ===============================
 # = SPRING PROFILE
 # ===============================
-spring.profiles.active=dev
+spring.profiles.active=local
```
Comment (krzjez) [inline src/main/resources/application.properties]: ?

File: src/main/java/com/metabug/validation/password/PasswordConstraintValidator.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public boolean isValid(final String password, final ConstraintValidatorContext c
         context.buildConstraintViolationWithTemplate(Joiner.on(",")
                 .join(validator.getMessages(result)))
                 .addConstraintViolation();
+
```
Comment (krzjez) [inline src/main/java/com/metabug/validation/password/PasswordConstraintValidator.java]: ?

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class EmailValidatorTest {
+
+    private static ConstraintValidatorContext validatorContext;
+
+    @BeforeAll
+    public static void setup() {
+
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: redundant empty line

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class EmailValidatorTest {
+
+    private static ConstraintValidatorContext validatorContext;
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: final?

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+        validatorContext = mock(ConstraintValidatorContext.class);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Test-Email123@gmail.com", "TestEmail123@mail.net", "testemail123@wp.pl", "Test.Email@o2.org", "96420@stud.uz.zgora.pl", "96_420@stud.uz.zgora.pl"})
+    public void shouldReturnTrueWhenEmailIsValid(String input) {
+        final EmailValidator emailValidator = new EmailValidator();
+
+        Boolean result = emailValidator.isValid(input, validatorContext);
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: final

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    @BeforeAll
+    public static void setup() {
+
+        validatorContext = mock(ConstraintValidatorContext.class);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Test-Email123@gmail.com", "TestEmail123@mail.net", "testemail123@wp.pl", "Test.Email@o2.org", "96420@stud.uz.zgora.pl", "96_420@stud.uz.zgora.pl"})
+    public void shouldReturnTrueWhenEmailIsValid(String input) {
+        final EmailValidator emailValidator = new EmailValidator();
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java:21]: add //given above

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: //when

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        Boolean result = emailValidator.isValid(input, validatorContext);
+
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: //then

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        assertTrue(result);
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"TestEmail1!!@gmail.com", "Test Email123@mail.net", "testemail123wp.pl", "TestEmail@o2org", "96=420@stud.uz.zgora.pl"})
+    public void shouldReturnFalseWhenEmailIsInvalid(String input) {
+
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: redundant empty line

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        Boolean result = emailValidator.isValid(input, validatorContext);
+
+        assertTrue(result);
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"TestEmail1!!@gmail.com", "Test Email123@mail.net", "testemail123wp.pl", "TestEmail@o2org", "96=420@stud.uz.zgora.pl"})
+    public void shouldReturnFalseWhenEmailIsInvalid(String input) {
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: final String input?

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+        //given
+        final EmailValidator emailValidator = new EmailValidator();
+
+        //then
+        Boolean result = emailValidator.isValid(input, validatorContext);
```
Comment (krzjez) [inline src/test/java/com/metabug/EmailValidatorTest.java]: final

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+public class PasswordConstraintValidatorTest {
+
+    private ConstraintValidatorContext validatorContext;
+    private ConstraintValidatorContext.ConstraintViolationBuilder constraintViolationBuilder;
+
+
+    @Test
+    public void shouldReturnTrueWhenPasswordIsValid() {
+
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: redundant empty line

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+        //given
+        final PasswordConstraintValidator passwordConstraintValidator = new PasswordConstraintValidator();
+
+       //when
+        validatorContext = mock(ConstraintValidatorContext.class);
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: move to BeforeAll

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!1Aaaaaaa", validatorContext);
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: final

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+    @Test
+    public void shouldReturnFalseWhenPasswordIsTooShort() {
+
+        //given
+        final PasswordConstraintValidator passwordConstraintValidator = new PasswordConstraintValidator();
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: I think that this should be in beforeAll

File: src/test/java/com/metabug/PasswordMatchesValidatorTest.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class PasswordMatchesValidatorTest {
+
+    private static ConstraintValidatorContext validatorContext;
+    private UserDto user;
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordMatchesValidatorTest.java]: final?

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+
+public class PasswordConstraintValidatorTest {
+
+    private ConstraintValidatorContext validatorContext;
+    private ConstraintValidatorContext.ConstraintViolationBuilder constraintViolationBuilder;
```
Comment (krzjez) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: final?

**Review by krzjez — CHANGES_REQUESTED** [review]

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password must be 8 or more characters in length.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!1Aa", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: `final`

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password must contain 1 or more lowercase characters.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!1AAAAAAA", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: `final`

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password must contain 1 or more uppercase characters.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!1aaaaaaa", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: `final`

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password must contain 1 or more digit characters.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!!Aaaaaaa", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java:75]: `final`

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password must contain 1 or more special characters.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("11Aaaaaaa", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java:88]: `final`

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+
+        //when
+        validatorContext = mock(ConstraintValidatorContext.class);
+        constraintViolationBuilder = mock(ConstraintValidatorContext.ConstraintViolationBuilder.class);
+        doNothing().when(validatorContext).disableDefaultConstraintViolation();
+        when(validatorContext.buildConstraintViolationWithTemplate("Password contains a whitespace character.")
+        ).thenReturn(constraintViolationBuilder);
+
+        //then
+        Boolean result = passwordConstraintValidator.isValid("!1Aaa aaaa", validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: `final`

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+
+    @BeforeAll
+    public static void setup() {
+
+        validatorContext = mock(ConstraintValidatorContext.class);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Test-Email123@gmail.com", "TestEmail123@mail.net", "testemail123@wp.pl", "Test.Email@o2.org", "96420@stud.uz.zgora.pl", "96_420@stud.uz.zgora.pl"})
+    public void shouldReturnTrueWhenEmailIsValid(String input) {
```
Comment (kujawad) [inline src/test/java/com/metabug/EmailValidatorTest.java]: `final`

File: src/test/java/com/metabug/EmailValidatorTest.java (hunk 11 above)
Comment (kujawad) [inline src/test/java/com/metabug/EmailValidatorTest.java]: You're testing `isValid` method, this should be in `//when` section

File: src/test/java/com/metabug/EmailValidatorTest.java [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"TestEmail1!!@gmail.com", "Test Email123@mail.net", "testemail123wp.pl", "TestEmail@o2org", "96=420@stud.uz.zgora.pl"})
+    public void shouldReturnFalseWhenEmailIsInvalid(String input) {
+
+        //given
+        final EmailValidator emailValidator = new EmailValidator();
+
```
Comment (kujawad) [inline src/test/java/com/metabug/EmailValidatorTest.java]: Missing `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 14 above)
Comment (kujawad) [inline src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: This should be in `//when` section. You're testing `isValid` method

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 18 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 19 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 20 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 21 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java:75]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 22 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java:88]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordConstraintValidatorTest.java (hunk 23 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordConstraintValidatorTest.java]: Should be in `//when` section

File: src/test/java/com/metabug/PasswordMatchesValidatorTest.java [hunk 26]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Test
+    public void shouldReturnTrueWhenPasswordsMatch() {
+
+        //given
+        final PasswordMatchesValidator passwordMatchesValidator = new PasswordMatchesValidator();
+
+        //when
+        user.setPassword("!Password123");
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordMatchesValidatorTest.java:28]: These should be in `//given` section, you're preparing object to testing

File: src/test/java/com/metabug/PasswordMatchesValidatorTest.java [hunk 27]
```diff
@@ -0,0 +1,61 @@
...
+        user.setMatchingPassword("!Password123");
+
+        //then
+        boolean result = passwordMatchesValidator.isValid(user, validatorContext);
```
Comment (kujawad) [inline src/test/java/com/metabug/PasswordMatchesValidatorTest.java]: 1. `final`
2. Should be in `//when` section

File: src/test/java/com/metabug/PasswordMatchesValidatorTest.java (hunk 26 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordMatchesValidatorTest.java:28]: Apply this in `shouldReturnFalseWhenPasswordsDontMatch` too

File: src/test/java/com/metabug/PasswordMatchesValidatorTest.java (hunk 27 above)
Comment (kujawad) [reply src/test/java/com/metabug/PasswordMatchesValidatorTest.java]: Apply this in `shouldReturnFalseWhenPasswordsDontMatch` too

**Review by kujawad — CHANGES_REQUESTED** [review]

**Pull request merged**